dealiiadapter.cc

```diff
@@ -770,6 +791,124 @@ void ElasticProblem<dim>::output_results(const unsigned int timestep) const
     std::cout<< "\t Output written to solution-" + std::to_string(timestep) + ".vtk \n" <<std::endl;
 }
 
+template <int dim>
+void ElasticProblem<dim>::configure_precice()
+{
+    precice.configure(parameters.config_file);
+    meshID = precice.getMeshID(parameters.mesh);
+    read_data_ID  = precice.getDataID(parameters.read_data_name, meshID);
+    write_data_ID = precice.getDataID(parameters.write_data_name, meshID);
+}
+
+template <int dim>
+void ElasticProblem<dim>::initialize_precice()
+{
+    //get n_interface_nodes
+    std::set<types::boundary_id> couplingBoundary;
+    couplingBoundary.insert(parameters.interface_mesh_ID);
+    const FEValuesExtractors::Scalar x_displacement(0);
+
+    DoFTools::extract_boundary_dofs(dof_handler, fe.component_mask(x_displacement), coupling_dofs, couplingBoundary);
+    n_interface_nodes = coupling_dofs.n_elements();//TODO: Adapt
+
+    write_data = new double [dim * n_interface_nodes];
+    read_data  = new double [dim * n_interface_nodes];
+    interface_nodes_positions = new double [dim*n_interface_nodes];
+    interface_nodes_IDs       = new int [n_interface_nodes];
+
+    // get nodes_positions
+    std::map<types::global_dof_index, Point<dim>> support_points;
+    DoFTools::map_dofs_to_support_points(MappingQ1<dim>(), dof_handler, support_points);
+    int position_iterator=0;
+    for (IndexSet::ElementIterator element=coupling_dofs.begin(); element!=coupling_dofs.end(); element++)
+    {
+        for(int jj=0; jj<dim; jj++)
+            interface_nodes_positions[position_iterator * dim + jj] = support_points[*element][jj];
```

Comment:
Are you absolutely sure that `support_points` includes all the `*element` keys?
If no, please use `support_points.find()` as `operator[]` inserts a new element if the key doesn't exist.

Reply:
Yes, the previously called function `map_dofs_to_support_points` assures this

Reply:
Good. Can you add a comment to document this please.